CASMTRIAGE-6432 add goss_test_fails_with_connection_refused to known issues docs (#4610)
Cray-HPE/docs-csm: troubleshooting/known_issues/goss_tests_fails_with_connection_refused.md

# Goss Test Fails with Connection Refused

## Table of contents

- [Introduction](#introduction)
- [Example Error](#example-error)
- [Examining the problem](#examining-the-problem)
- [Resolution](#resolution)
- [Other possible causes of this error](#other-possible-causes-of-this-error)

## Introduction

Follow this procedure when a Goss test fails with `Failed to establish a new connection: [Errno 111] Connection refused`.

## Example Error

> An example of this error can be seen below.

Reviewer: ```suggestion
The following is an example of this error.
```

ERROR: Skipping http://ncn-s003.hmn:8997/ncn-healthcheck-storage due to error
ERROR: Error encountered running http://ncn-s003.hmn:8999/ncn-afterpitreboot-healthcheck-storage tests: Unexpected error attempting GET request to http://ncn-s003.hmn:8999/ncn-afterpitreboot-healthcheck-storage: ConnectionError: HTTPConnectionPool(host='ncn-s003.hmn', port=8999):
Max retries exceeded with url: /ncn-afterpitreboot-healthcheck-storage (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f88902ecd68>: Failed to establish a new connection: [Errno 111] Connection refused',))
ERROR: Skipping http://ncn-s003.hmn:8999/ncn-afterpitreboot-healthcheck-storage due to error
```

## Examining the problem

This error means that not all Goss tests on the NCN are available via the Goss servers. This can be seen by comparing the output from
`systemctl status goss-servers` on multiple nodes of the same type (i.e. storage, master, worker). The node with the error will

> have less entries under `CGroup`. An example of output from a node with and without this error can be seen below.

Reviewer: ```suggestion
have less entries under `CGroup`. The following is an example of output from a node with and without this error.
```
